### PR TITLE
204 and 304 must not have a body, see https://tools.ietf.org/html/rfc…

### PR DESCRIPTION
**Pull request by blizzz**
…7230#section-3.3

from https://github.com/owncloud/core/pull/25835

@nickvergessen @MorrisJobke you have had concerns in the original PR

### CODE REVIEW

File: lib/public/AppFramework/Http/JSONResponse.php [hunk 1]
```diff
@@ -61,15 +61,21 @@ public function __construct($data=array(), $statusCode=Http::STATUS_OK) {
...
 	 */
 	public function render() {
-		$response = json_encode($this->data, JSON_HEX_TAG);
-		if($response === false) {
-			throw new \Exception(sprintf('Could not json_encode due to invalid ' .
-				'non UTF-8 characters in the array: %s', var_export($this->data, true)));
+		if ($this->getStatus() === Http::STATUS_NO_CONTENT
+			|| $this->getStatus() === Http::STATUS_NOT_MODIFIED
+		) {
+			$response = null;
```
Comment (LukasReschke) [inline lib/public/AppFramework/Http/JSONResponse.php]: I'd say we should only take this for master and then throw an exception. Don't do magic.

cc @nickvergessen @MorrisJobke

**Comment by rullzer** [comment]
This fix is the wrong way around.
Because lets say I set the right etag. And then a request is made with the IF-NONE-MATCH header that matches. Then the response doesn't know this.

So :-1:

**Comment by rullzer** [comment]
Also it is weird that it only does this for 1 response type

**Comment by blizzz** [comment]
Then, let's :fire: it

**Pull request closed without merging**